### ISSUE TITLE
LYWSD03MMC relevant data from connection get replaced by broadcast data

### ISSUE THREAD

**Issue by fhb** [1technophile/OpenMQTTGateway#980]
**Describe the bug**
Although I set the esp32 to connect with every scan the temperature/humidity/battery data on mosquitto (version 1.5.7) is getting replaced by the data broadcasted by the LYWSD03MMC sensor on a scan: 
*Broadcast data:*
```
{"id":"A4:C1:38:17:5F:77","name":"LYWSD03MMC","rssi":-40,"servicedata":"30585b055a775f1738c1a408"}
```
*Relevant data:*
```
{"model":"LYWSD03MMC","id":"A4:C1:38:17:5F:77","tempc":26.35,"tempf":79.43,"hum":56,"volt":2.919,"batt":81.9}
```
![image](https://user-images.githubusercontent.com/680408/121452711-2ead6d00-c9a0-11eb-911f-9edee34139d1.png)

Therefore the relevant data is not available to openhab or mqttthing on homebridge most of the time and errors are thrown. Also "onlysensors" and "whitelist" seem not to work properly (set through mqtt commands).

**To Reproduce**

- Hardware ESP32-WROOM-32, 3x LYWSD03MMC (with 1.0.0._0130 fw)
- Download OpenmQTTGateway Sources and load esp32dev_ble into the the arduino libraries
- Correct known bug on line 523 (missing function void....)
- #define OpenHABDiscovery true
- Install OpenMQTTGateway 0.9.6 on ESP32 dev with activated ZGateway_BLE through ArduinoIDE.
- Configure MQTT and WIFI
- set: scan to connect ratio to 1 ``mosquitto_pub  -t home/OpenMQTTGateway/commands/MQTTtoBT/config -m '{"scanbcnct":1}'``
- whitelist Sensors ``mosquitto_pub -t home/OpenMQTTGateway/commands/MQTTtoBT/config -m '{"white-list":[your MAC adresses]}'``
- trigger scan and hopefully connect too with ``mosquitto_pub -t home/OpenMQTTGateway/commands/MQTTtoBT/config -m '{"interval":0}'``
- also tried with ``mosquitto_pub -t home/OpenMQTTGateway/commands/MQTTtoBT/config -m '{"onlysensors":true}'``

**Expected behavior**
I would expect OpenMQTTGateway to only plublish the relevant data set after a successful connection and otherwise not push an mqtt message.

**Environment (please complete the following information):**
* Hardware ESP32-WROOM-32, 3x LYWSD03MMC (with 1.0.0._0130 fw)
* OpenMQTTGateway V0.9.6 with esp32dev_ble, most recent ArduinoIDE, MQTT-Broker Mosquitto V 1.5.7, 

**Possible workaround, earlier discussion:**
A workaround solution would be a broadcasting firmware. Similar issues seem to have been discussed here: https://community.openmqttgateway.com/t/xiaomi-mijia-lywsd03mmc-esp32-support/775/49

**Comment by 1technophile** [comment]
Would it be possible to share what you get on the serial monitor after the scan?

> Only solution would be a broadcasting firmware right now. Similar issues seem to have been discussed here:

This solution is better in terms of battery consumption and quite easy to implement (less than 5min to flash the ATC firmware)

**Comment by fhb** [comment]
Sure. It might be the connection failing from time to time, but I don't understand the serial output fully:

[omg_serialoutput.txt](https://github.com/1technophile/OpenMQTTGateway/files/6628970/omg_serialoutput.txt)


Even so it works with homebridge mqtt and it probably should work with openhab with the retain option selected.

**Comment by NorthernMan54** [comment]
I have a similar symptom with the DT24 Bluetooth voltmeter I added, and I found that two different message's were being sent on the same device topic.  I found that I was seeing a message from the BLE scan and a second different message from the device BT connect.

But I found that it did not cause an issue with my client as it was using home assistant mqtt discovery and it was ignoring the messages with missing fields.  Does openHab have something similar to the home assistant value template and the is_defined macro ?  In my setup that caused the messages to be ignored by the client.

```
{"stat_t":"+/+/BTtoMQTT/200507161361",
"name":"DT24-volt",
"uniq_id":"200507161361-DT24-volt",
"dev_cla":"power",
"val_tpl":"{{ value_json.volt | is_defined }}",
"unit_of_meas":"V",
"device":{"connections":[["mac","200507161361"]],"identifiers":["200507161361"],"manufacturer":"ATorch","model":"DT24-BLE","name":"DT24","via_device":"cc1101-dddddd"}}
```
